examples/step-19/doc/intro.dox

```diff
@@ -0,0 +1,352 @@
+@f[
+  \Delta t
+  \le
+  \min_i
+  \sqrt{ \frac{h_i m}{e\|\nabla V^{(1)}\| }.
+@f]
+Using the same argument about neighboring cells possibly being smaller by
+a factor of two then leads to the final formula for time step zero:
+@f[
+  \Delta t
+  =
+  \min_i
+  \sqrt{ \frac{\frac 12 h_i m}{e\|\nabla V^{(1)}\| }.
+@f]
+
+Strictly speaking, we would have to evaluate the electric potential $V^{(1)}$ at
+the location of each particle, but a good enough approximation is to use the
+maximum of the values at the vertices of the respective cell. (Why the vertices
+and not the midpoint? Because the gradient of the solution of the Laplace equation,
+i.e., the electric field, is largest in corner singularities which are located
+at the vertices of cells.) This has the advantage that we can make good use of the
+FEValues functionality which can recycle pre-computed material as long as the
+quadrature points are the same from one cell to the next.
+
+We could always run this kind of scheme to estimate the difference between
+$\mathbf v_i^{(n-1)}$ and $\mathbf v_i^{(n)}$, but it relies on evaluating the
+electric field $\mathbf E$ on each cell, and that is expensive. As a
+consequence, we will limit this approach to the very first time step.
+
+
+<h3>Spatial discretization</h3>
+
+Having discussed the time discretization, the discussion of the spatial
+discretization is going to be short: We use quadratic finite elements,
+i.e., the space $Q_2$, to approximate the electric potential $V$. The
+mesh is adapted a couple of times during the initial time step. All
+of this is entirely standard if you have read step-6, and the implementation
+does not provide for any kind of surprise.
+
+
+
+<h3>Dealing with particles programmatically</h3>
+
+Adding and moving particles is, in practice, not very difficult in deal.II.
+To add one, the `create_particles()` function of this program simply
+uses a code snippet of the following form:
+@code
+  Particles::Particle<dim> new_particle;
+  new_particle.set_location(location);
+  new_particle.set_reference_location
+      (mapping.transform_real_to_unit_cell(cell, location));
+  new_particle.set_id(n_current_particles);
+  
+  particle_handler.insert_particle(new_particle, cell);
+@endcode
+In other words, it is not all that different from inserting an object
+into a `std::set` or `std::map`: Create the object, set its properties
+(here, the current location, its reference cell location, and its id)
+and call `insert_particle`. The only thing that may be surprising is
+the reference location: In order to evaluate things such as 
+$\nabla V(\mathbf x_i)$, it is necessary to evaluate finite element
+fields at locations $\mathbf x_i$. But this requires evaluating the
+finite element shape functions at points on the refence cell 
+$\hat{\mathbf x}_i$. To make this efficient, every particle doesn't
+just store its location and the cell it is on, but also what location
+that point corresponds to reference coordinates.
+
```

Comment:
Maybe you could add a sentence here that describes that there are convenience functions for creating more complex particle patterns available in the `Particles::Generators` namespace, which will be illustrated in the other particle tutorials?

Reply:
I'm putting this into a comment at some point below. Thanks for the suggestion.